src/components/forms/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback } from 'react'
+import { View, TouchableWithoutFeedback } from 'react-native'
+import { useStyles } from '../../../theme'
+import { Box } from '../../structure'
+import { BodyText } from '../../text'
+import { shameStyles } from '../../../theme/shame-styles'
+
+export interface RadioButtonProps {
+  /**
+   * Label to display next to the Radio Button.
+   */
+  label: string
+  /**
+   * True if selected.
+   */
+  checked: boolean
+  /**
+   * Called when selection state changes. Should propagate change to `checked` prop.
+   */
+  onChange: (checked: boolean) => void
+  /**
+   * Additional text to aid in use.
+   */
+  helpText?: string
+}
+
+const { size, checkSize } = shameStyles.radioButton
+
+/**
+ * Use when the user needs to choose a single item in a list.
+ */
+export const RadioButton: React.FC<RadioButtonProps> = ({ label, checked, onChange, helpText }) => {
+  const styles = useStyles(theme => ({
+    background: {
+      backgroundColor: theme.colors.fill.background.lighter,
+      borderColor: theme.colors.fill.primary.default,
+      borderWidth: theme.border.small,
+      borderRadius: theme.radius.circle,
+      width: size,
+      height: size,
+      marginVertical: (theme.font.size.regular.height - size) / 2,
+      justifyContent: 'center',
+      alignItems: 'center',
+    },
+    backgroundChecked: {
+      borderColor: theme.colors.fill.accent.default,
+    },
+    checkmark: {
+      height: checkSize,
+      width: checkSize,
+      borderRadius: theme.radius.circle,
+      backgroundColor: theme.colors.fill.accent.default,
+      opacity: 0,
+    },
+    checkmarkChecked: {
+      opacity: 1,
+    },
```

Comment:
This could be possible to do. However, Stylesheets should be as static as possible and this would require to pass a dependency array to `useStyles` with `checked` in it.

Current approach is more aligned with current `Stylesheet` usage everywhere, while second option is more aligned with the Hooks mentalisty.

Honestly I don't have any preference between the 2. Defining multiple styles is maybe less error-prone than passing dependencies to `useStyles`, especially if there are multiple of them. Also, if animating the property, the property would be defined via an Animated property, which would prevent the need to have 2 separate styles.

Another solution might be to drop the memoization of the styles object, but I think this could have an impact on performance, as a new styles object will be passed each time to the view, therefore probably causing unnecessary logic/view rendering.